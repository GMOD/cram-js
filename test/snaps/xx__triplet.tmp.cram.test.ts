import { testFile } from '../testUtil'
testFile('xx#triplet.tmp.cram')
