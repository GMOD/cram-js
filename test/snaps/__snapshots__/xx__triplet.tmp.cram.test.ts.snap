// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole xx#triplet.tmp.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 45,
      "_size": 19,
      "alignmentSpan": 0,
      "crc32": 112725797,
      "landmarks": [
        0,
        132,
      ],
      "length": 207,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 45,
        "_size": 19,
        "alignmentSpan": 0,
        "crc32": 112725797,
        "landmarks": [
          0,
          132,
        ],
        "length": 207,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "yy",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "yy",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 17,
          },
          "parametersBytes": 1,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 15,
          },
          "parametersBytes": 1,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 21,
          },
          "parametersBytes": 1,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 23,
          },
          "parametersBytes": 1,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              10,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 22,
          },
          "parametersBytes": 1,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 270,
      "_size": 18,
      "alignmentSpan": 20,
      "crc32": 1439931990,
      "landmarks": [
        160,
      ],
      "length": 452,
      "numBases": 30,
      "numBlocks": 10,
      "numLandmarks": 1,
      "numRecords": 3,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 270,
        "_size": 18,
        "alignmentSpan": 20,
        "crc32": 1439931990,
        "landmarks": [
          160,
        ],
        "length": 452,
        "numBases": 30,
        "numBlocks": 10,
        "numLandmarks": 1,
        "numRecords": 3,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 67,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 6,
              "flags": 0,
              "readName": undefined,
              "sequenceId": 0,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "a1",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 20,
            "uniqueId": 436,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 35,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 11,
              "flags": 1,
              "readName": undefined,
              "sequenceId": 0,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "a1",
            "sequenceId": 0,
            "tags": {},
            "templateSize": -20,
            "uniqueId": 437,
          },
          {
            "alignmentStart": 11,
            "cramFlags": 3,
            "flags": 147,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "flags": 0,
              "readName": undefined,
              "sequenceId": 0,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "a1",
            "sequenceId": 0,
            "tags": {},
            "templateSize": -20,
            "uniqueId": 438,
          },
        ],
        "header": {
          "_endPosition": 474,
          "_size": 39,
          "compressedSize": 35,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            20,
            3,
            0,
            8,
            7,
            11,
            12,
            15,
            17,
            21,
            22,
            23,
            255,
            255,
            255,
            255,
            15,
            187,
            244,
            222,
            109,
            132,
            151,
            161,
            25,
            221,
            166,
            224,
            116,
            82,
            22,
            67,
            220,
          ],
          "contentId": 0,
          "contentPosition": 435,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 3913401814,
          "parsedContent": {
            "_endPosition": 305,
            "_size": 35,
            "contentIds": [
              11,
              12,
              15,
              17,
              21,
              22,
              23,
            ],
            "md5": [
              187,
              244,
              222,
              109,
              132,
              151,
              161,
              25,
              221,
              166,
              224,
              116,
              82,
              22,
              67,
              220,
            ],
            "numBlocks": 8,
            "numContentIds": 7,
            "numRecords": 3,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 20,
            "refSeqStart": 1,
          },
          "uncompressedSize": 35,
        },
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 17,
          },
          "parametersBytes": 1,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 15,
          },
          "parametersBytes": 1,
        },
        "CF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 16,
          },
          "parametersBytes": 1,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              10,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 740,
      "_size": 18,
      "alignmentSpan": 15,
      "crc32": 3505874123,
      "landmarks": [
        142,
      ],
      "length": 352,
      "numBases": 20,
      "numBlocks": 8,
      "numLandmarks": 1,
      "numRecords": 2,
      "recordCounter": 3,
      "refSeqId": 1,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 740,
        "_size": 18,
        "alignmentSpan": 15,
        "crc32": 3505874123,
        "landmarks": [
          142,
        ],
        "length": 352,
        "numBases": 20,
        "numBlocks": 8,
        "numLandmarks": 1,
        "numRecords": 2,
        "recordCounter": 3,
        "refSeqId": 1,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 67,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 6,
              "readName": "a1",
              "sequenceId": 1,
              "uniqueId": 892,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "a1",
            "sequenceId": 1,
            "tags": {},
            "templateLength": 15,
            "templateSize": undefined,
            "uniqueId": 891,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 1,
            "flags": 3,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "a1",
              "sequenceId": 1,
              "uniqueId": 891,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "a1",
            "sequenceId": 1,
            "tags": {},
            "templateLength": 15,
            "templateSize": undefined,
            "uniqueId": 892,
          },
        ],
        "header": {
          "_endPosition": 924,
          "_size": 37,
          "compressedSize": 33,
          "compressionMethod": "raw",
          "content": Uint8Array [
            1,
            1,
            15,
            2,
            3,
            6,
            5,
            11,
            12,
            15,
            16,
            17,
            255,
            255,
            255,
            255,
            15,
            27,
            49,
            18,
            78,
            242,
            247,
            161,
            116,
            5,
            146,
            240,
            144,
            216,
            197,
            17,
            225,
          ],
          "contentId": 0,
          "contentPosition": 887,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 2281897814,
          "parsedContent": {
            "_endPosition": 773,
            "_size": 33,
            "contentIds": [
              11,
              12,
              15,
              16,
              17,
            ],
            "md5": [
              27,
              49,
              18,
              78,
              242,
              247,
              161,
              116,
              5,
              146,
              240,
              144,
              216,
              197,
              17,
              225,
            ],
            "numBlocks": 6,
            "numContentIds": 5,
            "numRecords": 2,
            "recordCounter": 3,
            "refBaseBlockId": -1,
            "refSeqId": 1,
            "refSeqSpan": 15,
            "refSeqStart": 1,
          },
          "uncompressedSize": 33,
        },
      },
    ],
  },
]
`;
